pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisRecord.java

```diff
@@ -40,21 +41,30 @@
     private final byte[] value;
     private final HashMap<String, String> userProperties = new HashMap<String, String> ();
     
-    public KinesisRecord(com.amazonaws.services.kinesis.model.Record record) {
-        this.key = Optional.of(record.getPartitionKey());
-        setProperty(ARRIVAL_TIMESTAMP, record.getApproximateArrivalTimestamp().toString());
-        setProperty(ENCRYPTION_TYPE, record.getEncryptionType());
-        setProperty(PARTITION_KEY, record.getPartitionKey());
-        setProperty(SEQUENCE_NUMBER, record.getSequenceNumber());
-        
-        if (StringUtils.isBlank(record.getEncryptionType())) {
+    public KinesisRecord(KinesisClientRecord record) {
+        this.key = Optional.of(record.partitionKey());
+        // encryption type can (annoyingly) be null, so we default to NONE
+        EncryptionType encType = EncryptionType.NONE;
+        if (record.encryptionType() != null) {
+            encType = record.encryptionType();
+        }
+        setProperty(ARRIVAL_TIMESTAMP, record.approximateArrivalTimestamp().toString());
+        setProperty(ENCRYPTION_TYPE, encType.toString());
+        setProperty(PARTITION_KEY, record.partitionKey());
+        setProperty(SEQUENCE_NUMBER, record.sequenceNumber());
+
+        if (encType == EncryptionType.NONE) {
             String s = null;
             try {
-                s = decoder.decode(record.getData()).toString();
+                s = decoder.decode(record.data()).toString();
             } catch (CharacterCodingException e) {
                // Ignore 
             }
             this.value = (s != null) ? s.getBytes() : null;
+        } else if (encType == EncryptionType.KMS) {
+            // use the raw encrypted value, let them handle it downstream
+            // TODO: support decoding KMS data here... should be fairly simple
```

Comment:
Do we want to document this as a known limitation for now? Not sure where that would even make sense to call out.